langchain/src/llms/bittensor.ts

```diff
@@ -0,0 +1,179 @@
+import { BaseLLMParams, LLM } from "./base.js";
+
+export interface BittensorInput extends BaseLLMParams {
+  systemPrompt?: string | null | undefined;
+  topResponses?: number | undefined;
+}
+interface Message {
+  role: string;
+  content: string;
+}
+
+interface ChatPayload {
+  uids?: string[];
+  top_n?: number;
+  messages: Message[];
+}
+
+interface APIKeyResponse {
+  api_key: string;
+}
+
+interface ChatResponse {
+  choices?: { message: Message }[];
+}
+/**
+ * Class representing the Neural Internet language model powerd by Bittensor, a decentralized network
+ * full of different AI models.
+ * To analyze API_KEYS and logs of you usage visit
+ *      https://api.neuralinternet.ai/api-keys
+ *      https://api.neuralinternet.ai/logs
+ */
+export class NIBittensorLLM extends LLM implements BittensorInput {
+  systemPrompt: string | null | undefined;
+
+  topResponses: number | undefined;
+
+  constructor(fields?: BittensorInput) {
+    super(fields ?? {});
+    this.systemPrompt = fields?.systemPrompt;
+    this.topResponses = fields?.topResponses;
+  }
+
+  _llmType(): string {
+    return "NIBittensorLLM";
+  }
+
+  /** Call out to NIBittensorLLM's complete endpoint.
+   Args:
+       prompt: The prompt to pass into the model.
+
+       Returns: The string generated by the model.
+
+   Example:
+   let response = niBittensorLLM.call("Tell me a joke.");
+   */
+  async _call(prompt: string): Promise<string> {
+    const defaultPrompt =
```

Comment:
Why are these hard coded?

Reply:
I mean why are we hard-coding prompts here? Are they somehow required?

Reply:
I fixed that hard coded prompts.
Thanks.